drivers/include/mcp47xx.h

```diff
@@ -0,0 +1,347 @@
+/** @} */
+
+/**
+ * @brief   MCP47xx driver error codes */
+typedef enum {
+    MCP47XX_OK,                 /**< success */
+    MCP47XX_ERROR_I2C,          /**< I2C communication error */
+    MCP47XX_ERROR_NOT_AVAIL,    /**< device not available */
+} mcp47xx_error_codes_t;
+
+/**
+ * @brief   Supported MCP47xx variants
+ *
+ * It is used in configuration parameters to specify the MCP47xx expander
+ * used by device.
+ */
+typedef enum {
+    MCP4706,    /**< 1 channel 8-bit DAC */
+    MCP4716,    /**< 1 channel 10-bit DAC */
+    MCP4725,    /**< 1 channel 12-bit DAC */
+    MCP4726,    /**< 1 channel 12-bit DAC */
+    MCP4728,    /**< 4 channel 12-bit DAC */
+} mcp47xx_variant_t;
+
+/**
+ * @brief   MCP47xx gain configuration type
+ *
+ * @note Gains are not supported by MCP4725.
+ */
+typedef enum {
+    MCP47XX_GAIN_1X = 0,  /**< Gain is 1.0, supported by all MCP47xx variants */
+    MCP47XX_GAIN_2X = 1,  /**< Gain is 2.0, not supported by MCP4725 */
+} mcp47xx_gain_t;
+
+/**
+ * @brief   MCP47xx V_REF configuration type
+ *
+ * @note Different MCP47xx variants allow different V_REF configurations
+ */
+typedef enum {
+    MCP47XX_VREF_VDD = 0,   /**< V_REF = V_DD, supported by all MCP47xx */
+    MCP47XX_VREF_INT = 1,   /**< V_REF = internal (2.048 V), MCP4728 only  */
+    MCP47XX_VREF_PIN = 2,   /**< V_REF = VREF pin not buffered, MCP47x6 only */
+    MCP47XX_VREF_BUF = 3,   /**< V_REF = VREF pin buffered, MCP47x6 only */
+} mcp47xx_vref_t;
+
+/**
+ * @brief   MCP47xx Power-down mode selection type
+ *
+ * Defines the possible power-down modes used for MCP47xx device configuration.
+ * The mode is used by function #mcp47xx_dac_poweroff to set the DAC into the
+ * configured power-down mode.
+ *
+ * @note #MCP47XX_NORMAL cannot be configured as power-down mode.
+ */
+typedef enum {
+    MCP47XX_NORMAL = 0,     /**< Normal mode */
+    MCP47XX_PD_SMALL = 1,   /**< Power down, small resistor 1 kOhm */
+    MCP47XX_PD_MEDIUM = 2,  /**< Power down, medium resistor,
+                                 125 kOhm for MCP47x6, 100 kOhm otherwise */
+    MCP47XX_PD_LARGE = 3,   /**< Power down, large resistor,
+                                 640 kOhm for MCP47x6, 125 kOhm otherwise */
+} mcp47xx_pd_mode_t;
+
+/**
+ * @brief   MCP47xx device configuration parameters
+ */
+typedef struct {
+
+    i2c_t    dev;       /**< I2C device */
+    uint16_t addr;      /**< I2C slave address MCP47XX_BASE_ADDR + [0...7] */
+
+    mcp47xx_variant_t   variant;    /**< used variant of MCP47xx */
+    mcp47xx_gain_t      gain;       /**< Gain selection */
+    mcp47xx_vref_t      vref;       /**< Voltage reference selection */
+    mcp47xx_pd_mode_t   pd_mode;    /**< Power-down mode selection */
+
+} mcp47xx_params_t;
+
+/**
+ * @brief   MCP47xx device data structure type
+ */
+typedef struct {
+    mcp47xx_params_t params;              /**< device configuration parameters */
+    uint16_t values[MCP47XX_CHN_NUM_MAX]; /**< contains the last values set */
```

Comment:
```suggestion
    uint16_t values[MCP47XX_CHN_NUM_MAX]; /**< contains the last values set for persistence when device is powered off  */
```

right?

Reply:
Yes, wil change it. May I squash directly?

Reply:
Sure, squash